os/src-jvm/ProcessOps.scala

```diff
@@ -337,12 +337,17 @@ private[os] object ProcessOps {
   ): ProcessBuilder = {
     val builder = new java.lang.ProcessBuilder()
 
-    val baseEnv =
-      if (propagateEnv) sys.env
-      else Map()
-    for ((k, v) <- baseEnv ++ Option(env).getOrElse(Map())) {
-      if (v != null) builder.environment().put(k, v)
-      else builder.environment().remove(k)
+    val environment = builder.environment()
+
+    if (!propagateEnv) {
+      environment.clear()
```

Comment:
`java.lang.ProcessBuilder` propagates the environment by default. If you don't want to propagate it you need to `clear()` the map.

Reply:
Or even assert that it is set as expected.